core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -241,7 +241,7 @@ protected[core] object WhiskWebActionsApi extends Directives {
 
             fields.get("body") map {
                 case JsString(str) => interpretHttpResponse(code, headers, str, transid)
-                case _             => terminate(BadRequest, Messages.httpContentTypeError)(transid)
+                case js            => interpretHttpResponseAsJson(code, headers, js, transid)
```

Comment:
I'd have to match against js object, array, etc. don't think it's necessary - jit should get there.